typescript/static-site/static-site.ts

```diff
@@ -49,21 +69,36 @@ export class StaticSite extends Construct {
         }).certificateArn;
         new cdk.CfnOutput(this, 'Certificate', { value: certificateArn });
 
-        // CloudFront distribution that provides HTTPS
-        const distribution = new cloudfront.CloudFrontWebDistribution(this, 'SiteDistribution', {
-            aliasConfiguration: {
-                acmCertRef: certificateArn,
-                names: [ siteDomain ],
+        // Specifies you want viewers to use HTTPS & TLS v1.1 to request your objects
+        const viewerCertificate = cloudfront.ViewerCertificate.fromAcmCertificate({
+            certificateArn: certificateArn,
+            env: {
+                region: cdk.Aws.REGION,
+                account: cdk.Aws.ACCOUNT_ID
+            },
+            node: this.node,
+            stack: parent
+        },
```

Comment:
`ICertificate` requires `metricDaysToExpiry` method but that's not defined on this object. There for ​tsc fails with:

```
cdk/static-site.ts:72:83 - error TS2345: Argument of type '{ certificateArn: string; env: { region: string; account: string; }; node: cdk.ConstructNode; stack: cdk.Stack; }' is not assignable to parameter of type 'ICertificate'.
 ​Property 'metricDaysToExpiry' is missing in type '{ certificateArn: string; env: { region: string; account: string; }; node: cdk.ConstructNode; stack: cdk.Stack; }' but required in type 'ICertificate'.

​72         const viewerCertificate = cloudfront.ViewerCertificate.fromAcmCertificate({
                                                                                     ​~
​73                 certificateArn: certificateArn,
   ​~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
...
​79                 stack: parent,
   ​~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
​80             },
   ​~~~~~~~~~~~~~

 ​node_modules/@aws-cdk/aws-certificatemanager/lib/certificate.d.ts:28:5
   ​28     metricDaysToExpiry(props?: cloudwatch.MetricOptions): cloudwatch.Metric;
          ​~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   ​'metricDaysToExpiry' is declared here.


Found 1 error.
```

Should I open an issue?

Reply:
I had to add this to my Stack to fix the typescript issue:

```ts
import * as cloudwatch from "@aws-cdk/aws-cloudwatch";

// then on the error:

metricDaysToExpiry: () =>
    new cloudwatch.Metric({
        namespace: "some namespace",
         metricName: "TLS Viewer Certificade Expired",
      }),
```

Reply:
Thanks for flagging this. I've got a fix here: https://github.com/aws-samples/aws-cdk-examples/pull/479